crm-plugin/controllers/activity.js

```diff
@@ -0,0 +1,106 @@
+"use strict";
+
+/**
+ * crm-plugin.js controller
+ *
+ * @description: A set of functions called "actions" of the `crm-plugin` plugin.
+ */
+const { sanitizeEntity } = require("strapi-utils");
+module.exports = {
+  /**
+   * Default action.
+   *
+   * @return {Object}
+   */
+
+  index: async (ctx) => {
+    // Add your own logic here.
+    // Send 200 `ok`
+    ctx.send({
+      message: "ok",
+    });
+  },
+
+  find: async (ctx) => {
+    let activity;
+    if (ctx.query._q) {
+      console.log("sdfhjdsfhjfdshj", ctx.query._q);
+      activity = await strapi.query("activity", "crm-plugin").search(ctx.query);
+    } else {
+      activity = await strapi.query("activity", "crm-plugin").find(ctx.query);
+    }
+
+    return activity.map((entity) =>
+      sanitizeEntity(entity, {
+        model: strapi.plugins["crm-plugin"].models["activity"],
+      })
+    );
+  },
+
+  create: async (ctx) => {
+    let actType;
+    let activityassignee;
+    let activity;
+    try {
+      if (ctx.params.id) {
+        const { id } = ctx.params;
+        activity = await strapi
+          .query("activity", "crm-plugin")
+          .update({ id }, ctx.request.body);
+          if(ctx.request.body.contact){
+            let activityDetail = {
```

Comment:
order wrong. contact and activityassignee has to be updated first and then activity

Reply:
done

Reply:
done